network/gossip/libp2p/libp2pNode_test.go

```diff
@@ -332,75 +329,62 @@ func (l *LibP2PNodeTestSuite) TestCreateStreamTimeoutWithUnresponsiveNode() {
+	assert.Error(suite.T(), err)
 }
 
 // TestCreateStreamIsConcurrent tests that CreateStream calls can be made concurrently such that one blocked call
 // does not block another concurrent call.
-func (l *LibP2PNodeTestSuite) TestCreateStreamIsConcurrent() {
+func (suite *LibP2PNodeTestSuite) TestCreateStreamIsConcurrent() {
 
 	// bump up the unicast timeout to a high value
 	unicastTimeout = time.Hour
 
 	// create two regular node
-	goodPeers, goodAddrs := l.CreateNodes(2, nil, false)
-	defer l.StopNodes(goodPeers)
-	require.Len(l.T(), goodAddrs, 2)
+	goodPeers, goodAddrs := suite.CreateNodes(2, nil, false)
+	defer suite.StopNodes(goodPeers)
+	require.Len(suite.T(), goodAddrs, 2)
 
 	// create a silent node which never replies
-	listener, silentNodeAddress := newSilentNode(l.T())
-	defer listener.Close()
-
-	wg := sync.WaitGroup{}
-	wg.Add(1)
-
-	ch := make(chan struct{})
-	// spin off a go routine to create a stream to the unresponsive node
-	go func() {
-		wg.Done()
-		_, _ = goodPeers[0].CreateStream(l.ctx, silentNodeAddress) // this call will block
-		close(ch)
+	listener, silentNodeAddress := newSilentNode(suite.T())
+	defer func() {
+		require.NoError(suite.T(), listener.Close())
 	}()
 
-	// make sure the go routine to create a stream with the unresponsive node actually started
-	unittest.AssertReturnsBefore(l.T(), wg.Wait, 5*time.Millisecond)
-	// make sure the go routine to create a stream did not finish
-	select {
-	case <-time.After(10 * time.Millisecond):
-	case <-ch:
-		assert.Fail(l.T(), "CreateStream attempt to the unresponsive peer did not block")
-	}
+	// creates a stream to unresponsive node and makes sure that the stream creation is blocked
+	streamCreated := unittest.RequireNeverReturnBefore(suite.T(),
+		func() {
+			_, _ = goodPeers[0].CreateStream(suite.ctx, silentNodeAddress) // this call will block
+		},
+		1*time.Second,
+		"CreateStream attempt to the unresponsive peer did not block")
 
-	var err error
-	// assert that the same peer can still connect to the other regular peer without being blocked
-	unittest.AssertReturnsBefore(l.T(),
+	// requires same peer can still connect to the other regular peer without being blocked
+	unittest.RequireReturnsBefore(suite.T(),
 		func() {
-			_, err = goodPeers[0].CreateStream(l.ctx, goodAddrs[1])
+			_, err := goodPeers[0].CreateStream(suite.ctx, goodAddrs[1])
+			require.NoError(suite.T(), err)
 		},
-		10*time.Millisecond)
-	assert.NoError(l.T(), err)
+		1*time.Second, "creating stream to a good node")
 
-	// assert that the CreateStream call to the unresponsive node was blocked while we attempted the CreateStream to the
+	// requires the CreateStream call to the unresponsive node was blocked while we attempted the CreateStream to the
 	// good address
-	select {
-	case <-ch:
-		assert.Fail(l.T(), "CreateStream attempt to the unresponsive peer did not block")
-	default:
-	}
+	unittest.RequireNeverClosedWithin(suite.T(), streamCreated, 1*time.Millisecond,
```

Comment:
We set timeout to `1ms` since we evaluate that the channel should be _still_ open at this time. Here the casualty is important, i.e., the channel should remain open even after creating a stream to a good node.

Reply:
yup makes sense! - I would just rename streamCreated to `blockedCallCh` or something that reflects that it is a channel.